From 4ec002260e7a2477f6eed7381d4b5b97be29e8d4 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 18 Oct 2023 15:53:28 +0200
Subject: [PATCH 0034/1141] cpuidle: stm32: add sentinel to compatibles list

The list of compatibles in struct of_device_id must be terminated
by a sentinel.
Issue identified with kazan address sanitizer.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ie29aa84a8937527518811894b929e6c5db213776
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/339891
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/cpuidle/cpuidle-stm32.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/cpuidle/cpuidle-stm32.c
+++ b/drivers/cpuidle/cpuidle-stm32.c
@@ -257,6 +257,7 @@ static const struct of_device_id stm32_c
 	{
 		.compatible = "stm32,cpuidle",
 	},
+	{ /* sentinel */ },
 };
 
 static struct platform_driver stm32_cpuidle_driver = {
